### PR TITLE
Travis db testing

### PR DESCRIPTION
**Pull request by gartner**
Add travis testing on both mysql and postgresql, see #288 

- the tests are running now, but some fail on mysql, others fail on pgsql. Will look into that. @

### CODE REVIEW

**Comment by romaninsh** [comment]
This looks well. For those tests that verify the specific syntax with MySQL style escaping, simply add ignore for PostgreSQL.

File: tests/Structure.php [hunk 1]
```diff
@@ -14,6 +14,11 @@ class Structure extends Expression
         'drop'   => 'drop table if exists {table}',
     ];
 
+    public function setEscapeChar($char)
+    {
+        $this->escape_char = $char;
+    }
+
```
Comment (romaninsh) [inline tests/Structure.php]: this class is actually defunct and should me removed in favour of atk4/schema. I'll add a separate ticket for that.

Comment (gartner) [reply tests/Structure.php]: I saw the notes about Structure.php somewhere, but would just tackle one thing at a time. So I went with just getting it running.

Comment (romaninsh) [reply tests/Structure.php]: yes, that's good. I'm on vacation now but if i find some free time i'll add ignores to make your PR work and will merge.

**Comment by romaninsh** [comment]
I have added some skips for generated sql code into https://github.com/atk4/data/tree/pr/292. Still there are other errors.

**Comment by gartner** [comment]
I am currently looking into these errors. For postgresql, there are lots of problems with duplicate primary keys...

**Comment by gartner** [comment]
Some of the errors that appear when testing against postgresql, are there because of the difference in autu-numbering primary keys. MySQLs auto_increment behaves a little like using "SELECT max() + 1 as id', so if you manually sets an id for a row and insert it, the next row you try to insert without a set id, will be given the number that comes next.
Postgresql uses sequences, which are only incremented if they are actualy used. So inserting a row where you manually set the id to 1, and then the next row you have not set the id yourself, it will use the sequence. Which will now return1, as it is first time it is used. And then the test fails, id 1 is already used.

What is the most correct way to fix this? Should I just mark them as expected to fail when database is postgresql?

**Comment by DarkSide666** [comment]
Why not use new `$model->sequence` feature for postgresql?

**Comment by gartner** [comment]
First of, I haven't implemented the sequence-feature for postgresql. But I cannot image that to be a big problem, when I can look at how its done for Oracle.

But also, I am under the impression that atk4/dsql is to abstract away the features/quirks/differences on the backend databases. Therefore, in my mind it would be wrong to make different setups for this many of the tests, since it is not really what is tested. 

The sequence-feature could be a solution, but then that feature should also be found for mysql and sqlite, i think.

**Comment by romaninsh** [comment]
sequence is for explicit use of sequences. In this case it's probably good just to skip tests which are not bogus.

**Comment by romaninsh** [comment]
Tests seem to work now fine, so I'm going to merge this.

**Comment by romaninsh** [comment]
Another TODO: merge coverage reports, otherwise it seems that we're decreasing coverage rather than increasing.

**Pull request merged**